File: Pascal/Task225.pas
program Task225;
var m, f, fp, fn, n: integer;
begin
    fp := 1; fn := 1; {початкові члени послідовності}
    n := 2; {n - номер останнього числа послідовності}
    writeln('Введіть натуральне число:');
    readln(m);
    while fn < m do
    begin
        f := fp + fn;
        {переходимо до пари наступних чисел Фібоначчі}
        fp := fn; fn := f; n := n + 1
    end;
    if fn = m then
        writeln( m, ' є ', n, '-им числом Фібоначчі')
    else  writeln( 'Найближче до ', m, 
        ' справа число Фібоначчі: ', fn);
end.

File: Kotlin/Task225.kt
fun main() {
    println("Введіть натуральне число для перевірки: ")
    val m = readln().toInt()
    var fp = 1
    var fn = 1
    var n = 2 // номер останнього числа послідовності
    var f:Int;
    while (fn < m) {
        f = fp + fn
        fp = fn; fn = f
        n += 1
    }
    if (fn == m) println("%d є ".format(m) +
            "%d-им числом Фібоначчі".format(n))
    else println("Найближче до %d, ".format(m) +
            "справа число Фібоначчі: %d".format(fn))
}
